PLUB/Sources/Views/Home/ViewModel/RecruitmentFilterViewModel.swift

```diff
@@ -29,123 +27,203 @@ final class RecruitmentFilterViewModel: RecruitmentFilterViewModelType {
+                return RecruitmentFilterCollectionViewCellModel(model: model, isTapped: false)
+              }
+              return model
+            }
+            selectingSubCategories.accept(fetchList)
+            let confirmList = confirmList.filter { $0 != subCategoryID }
+            confirmSubCategory.accept(confirmList)
+          }
+        }
+      })
+      .disposed(by: disposeBag)
+    
+    // 요일을 선택함에 따른 동작
+    isSelectingDay
+      .subscribe(onNext: { selectInfo in
+        let (isSelect, day) = selectInfo
+        
+        var confirmList = confirmDay.value
+        let fetchList = fetchingDayModel.value
+        
+        if day == .all { // [요일무관] 요일을 선택/미선택 했을 때
+          if isSelect {
+            let fetchList = fetchList.map { model in
+              model.day == .all ? RecruitmentFilterDateCollectionViewCellModel(day: .all, isTapped: true) : RecruitmentFilterDateCollectionViewCellModel(day: model.day)
+            }
+            fetchingDayModel.accept(fetchList)
+            confirmDay.accept([])
+          }
+          else {
+            let fetchList = fetchList.map { model in
+              var model = model
+              if model.day == .all {
+                model.isTapped.toggle()
+              }
+              return model
+            }
+            fetchingDayModel.accept(fetchList)
+          }
+        }
+        else { // 특정 요일을 선택/미선택 했을 때
+          if isSelect {
+            let fetchList = fetchList.map { model in
+              if model.day == day {
+                return RecruitmentFilterDateCollectionViewCellModel(day: model.day, isTapped: true)
+              }
+              else if model.day == .all {
+                return RecruitmentFilterDateCollectionViewCellModel(day: model.day)
+              }
+              return model
+            }
+            fetchingDayModel.accept(fetchList)
+            confirmList.append(day.eng)
+            confirmDay.accept(confirmList)
+          }
+          else {
+            let fetchList = fetchList.map { model in
+              if model.day == day {
+                return RecruitmentFilterDateCollectionViewCellModel(day: day)
+              }
+              return model
+            }
+            fetchingDayModel.accept(fetchList)
+            let confirmList = confirmList.filter { $0 != day.eng }
+            confirmDay.accept(confirmList)
+          }
+        }
+      })
+      .disposed(by: disposeBag)
     
     // Output
     selectedSubCategories = selectingSubCategories.asSignal(onErrorSignalWith: .empty())
     
     isButtonEnabled = Observable.combineLatest(
-      subCategoryCount,
-      dayCount
+      selectingSubCategories,
+      fetchingDayModel
     )
-    { subCategoryCnt, dayCnt -> Bool in
-      return subCategoryCnt != 0 && dayCnt != 0
+    { subCategories, days -> Bool in
+      return subCategories.filter { $0.isTapped }.count > 0 && days.filter { $0.isTapped }.count > 0
     }
```

Comment:
하나의 프로퍼티만을 처리한다면, KeyPath를 활용해보세요!
```suggestion
    { subCategories, days in
      return subCategories.filter(\.isTapped).count > 0 && days.filter(\.isTapped).count > 0
    }
```

Reply:
오 뭐야 이거 한번 사용해보도록 하겠습니다 감사합니다 👍